packages/client/components/TransactionDetails.tsx

```diff
@@ -0,0 +1,255 @@
+import React from "react";
+import { Text, HStack, Box, VStack, Badge, ScrollView } from "native-base";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+import { Pressable, StyleSheet } from "react-native";
+import { Transaction } from "@cryptify/common/src/domain/entities/transaction";
+import { faCircleArrowDownLeftCustom } from "./icons/faCircleArrowDownLeftCustom";
+import { faCircleArrowUpRightCustom } from "./icons/faCircleArrowUpRightCustom";
+import { faCopyCustom } from "./icons/faCopyCustom";
+import * as Clipboard from "expo-clipboard";
+
+type Props = {
+    transaction: Transaction;
+    walletAddress: string;
+};
+
+export function TransactionDetails({ transaction, walletAddress }: Props) {
+    const isIncommingTransaction = walletAddress == transaction.walletIn;
+
+    const copyToClipboard = async (valueToCopy: string) => {
+        await Clipboard.setStringAsync(valueToCopy);
+    };
+
+    function formatTransactionAddress(address: string): string {
+        return `${address.substring(0, 6)}...${address.substring(address.length - 4, address.length)}`;
+    }
+
+    function getCurrencyType(): string {
+        return transaction.transactionAddress.substring(0, 2) == "0x" ? "ETH" : "BTC";
+    }
+
+    function renderHeader() {
+        return (
+            <Box style={styles.itemWrapper}>
```

Comment:
I like your idea here but no need to make them functions because they don't depend on any parameters.

```suggestion
    const renderHeader = {
         <Box style={styles.itemWrapper}>
```